driver/pad_constant_driver.hpp

```diff
@@ -0,0 +1,389 @@
+
+    for(size_t gid = 0; gid < output_size; ++gid)
+    {
+        bool flag = true;
+        GET_NCDHW(o[0], o[1], o[2], o[3], o[4], gid, output_dims);
+
+        for(int i = 0; i < 5; i++)
+        {
+            o[i] = o[i] - padding[2 * i];
+            flag *= (o[i] < input_dims[i]);
+        }
+
+        if(flag)
+        {
+            output_host[gid] = get5DValueAt(input, input_dims.data(), o[0], o[1], o[2], o[3], o[4]);
+        }
+        else
+        {
+            output_host[gid] = value;
+        }
+    }
+}
+
+template <typename Tgpu, typename Tref>
+class ConstantPadDriver : public Driver
+{
+
+public:
+    ConstantPadDriver() : Driver()
+    {
+        miopenCreateTensorDescriptor(&inputDesc);
+        miopenCreateTensorDescriptor(&outputDesc);
+
+        data_type = miopen_type<Tgpu>{};
+    }
+
+    int AddCmdLineArgs() override;
+    int ParseCmdLineArgs(int argc, char* argv[]) override;
+    InputFlags& GetInputFlags() override { return inflags; }
+
+    int GetandSetData() override;
+    std::vector<int> GetInputTensorLengthsFromCmdLine();
+    std::vector<size_t> GetPaddingsFromCmdLine();
+
+    int AllocateBuffersAndCopy() override;
+
+    int RunForwardGPU() override;
+    int RunForwardCPU();
+
+    int RunBackwardGPU() override;
+    int RunBackwardCPU();
+
+    Tref GetTolerance();
+    int VerifyBackward() override;
+    int VerifyForward() override;
+    ~ConstantPadDriver() override
+    {
+        miopenDestroyTensorDescriptor(inputDesc);
+        miopenDestroyTensorDescriptor(outputDesc);
+    }
+
+private:
+    InputFlags inflags;
+
+    miopenTensorDescriptor_t inputDesc;
+    miopenTensorDescriptor_t outputDesc;
+
+    std::unique_ptr<GPUMem> in_dev;
+    std::unique_ptr<GPUMem> out_dev;
+    std::unique_ptr<GPUMem> padding_dev;
+
+    std::vector<Tgpu> input;
+    std::vector<Tgpu> output;
+    std::vector<Tref> output_host;
+
+    std::vector<size_t> padding;
+    Tgpu value;
+};
+
+template <typename Tgpu, typename Tref>
+int32_t ConstantPadDriver<Tgpu, Tref>::ParseCmdLineArgs(int argc, char* argv[])
+{
+    inflags.Parse(argc, argv);
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        miopenEnableProfiling(GetHandle(), true);
+    }
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int32_t ConstantPadDriver<Tgpu, Tref>::GetandSetData()
+{
+    padding = GetPaddingsFromCmdLine();
+
+    std::vector<int> input_dims = GetInputTensorLengthsFromCmdLine();
+    SetTensorNd(inputDesc, input_dims, data_type);
+
+    std::vector<int> output_dims;
+    output_dims.reserve(input_dims.size());
+
+    for(int i = 0; i < input_dims.size(); i++)
+    {
+        // TODO: ask the modnn team why it is like this.
+        output_dims.push_back(input_dims[i] + 2 * padding[2 * i]);
+    }
+
+    SetTensorNd(outputDesc, output_dims, data_type);
+
+    // Parse GetValueString -> (float)value
+    auto value_str = inflags.GetValueStr("value");
```

Comment:
Recommended) There is also a function called GetValueDouble.

Reply:
I’ve switched to using it. Thank you!